File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int count =0;
        int maxi =0;

        for(auto it: s)
        {
            if(it=='(')
            {
                count++;
                maxi = max(maxi, count);
            }
            else if(it ==')')
            {
                count--;
            }
        }
        return maxi;
        
    }
};

File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int count = 0;
        int n = s.length();
        int max  = 0;
        for(int i=0;i<n;i++)
        {
            char ch = s.charAt(i);

            if(ch == '(')
            {
                count++;
                max = Math.max(max, count);
            }
            else if(ch == ')')
            {
                count--;
            }
        }
        return max;
        
    }
}